code/modules/mob/living/carbon/xenomorph/XenoProcs.dm

```diff
@@ -261,6 +284,8 @@
 					spawn(charge_type == 1 ? 5 : 15)
 						frozen = FALSE
 						update_canmove()
+					if(stealth) //If we're stealthed, the stealth ends
```

Comment:
Add this in a `/mob/living/carbon/Xenomorph/Hunter/proc/update_progression()`

Reply:
? This will remove stealth on pouncing a target as intended?

Reply:
And instead of `if(stealth) //If we're stealthed, the stealth ends` you go with `if(check_stealth())`

There are other ways to do this, of course.

Reply:
The endgame is that hunter vars are only assigned to hunters, and that we don't have an `istype` check every time.